ui/uppy/src/components/PickerPanelTopBar.jsx

```diff
@@ -76,7 +76,30 @@ function UploadStatus({
 }
 
 function PanelTopBar(props) {
-  const { i18n, isAllComplete, hideCancelButton, maxNumberOfFiles, toggleAddFilesPanel, uppy } = props;
+  const {
+    i18n,
+    isAllComplete,
+    isAllErrored,
+    isAllPaused,
+    files,
+    invalidFiles,
+    newFiles,
+    hideCancelButton,
+    maxNumberOfFiles,
+    toggleAddFilesPanel,
+    uppy
+  } = props;
+  const uploadingState = getUploadingState(isAllErrored, isAllComplete, isAllPaused, files);
+  const invalidFilesCount = Object.values(invalidFiles).filter((invalid) => invalid).length;
+  const newFilesCount = newFiles.length;
+  const disableProceed =
+    // if there are active uploads
+    uploadingState === 'uploading' ||
+    // there are no new files to upload (all files already uploaded)
+    !newFilesCount ||
+    // if there are invalid files (due to site policies) waiting to be uploaded
+    (invalidFilesCount && uploadingState === 'waiting');
```

Comment:
- This looks like if files are added whilst other files are being uploaded, the proceed will be disabled?
- If newFilesCount is a number, let's express that condition as dealing with numbers (i.e. newFilesCount === 0)

Reply:
this comment is repeated, I'll resolve this and keep the other one until the details are actually resolved